DOL.WHD.Section14c.Web/src/modules/components/attachmentField/attachmentFieldController.test.js

```diff
@@ -29,34 +29,62 @@ describe('attachmentFieldController', function() {
     })
   );
 
-  it('attachment selected no files, file value should stay the same', function() {
+  it('should only allows specific file types: pdf jpg jpeg png csv PDF', function() {
+    var controller = attachmentFieldController();
+    var fileInput = { files: [{name: 'name1.docx'}] };
+    controller.onAttachmentSelected(fileInput);
+    expect(controller.upload.status).toBe('Invalid');
+    expect(controller.upload.message).toBe('Invalid File Type.');
+    expect(mockApiService.uploadAttachment).not.toHaveBeenCalled();
+  });
+
+  it('prevents multiple files from being uploaded', function() {
     var controller = attachmentFieldController();
-    var fileInput = { files: [], value: 1 };
+    var fileInput = { files: [{name: 'name1.pdf', size: 1000}] };
     controller.onAttachmentSelected(fileInput);
+    uploadAttachment.resolve({ data: [{ id: 1, originalFileName: 'name1.pdf' }] });
+    scope.$apply();
+    expect(scope.restrictUpload).toBe(true);
+    expect(controller.upload.status).toBe('Success');
+  });
 
+  it('it should prevent uploads of larger than 5MB', function() {
+    var controller = attachmentFieldController();
+    var fileInput = { files: [{name: 'name1.pdf', size: '10000000000'}] };
+    controller.onAttachmentSelected(fileInput);
+    expect(mockApiService.uploadAttachment).not.toHaveBeenCalled();
```

Comment:
Should this also test that `controller.upload.status` and `controller.upload.message` are set to `Invalid` and `File Size too large`, respectively?